cng/ecdh.go

```diff
@@ -194,16 +218,43 @@ func (k *PrivateKeyECDH) PublicKey() (*PublicKeyECDH, error) {
 	if err != nil {
 		return nil, err
 	}
-	consumeBigInt := func(size uint32) BigInt {
-		b := data[:size]
-		data = data[size:]
-		return b
-	}
-	X := consumeBigInt(hdr.KeySize)
```

Comment:
Was there a reason we were previously extracting the X and Y coords separately before appending them?

Reply:
The reason is that I initially copy-pasted this function from `GenerateKeyECDSA`, but that one returns the coordinates as separate entities while `GenerateKeyECDH` returns them as a single slice so there is no need to split `data`.